src/herder/test/TxSetTests.cpp

```diff
@@ -443,21 +482,36 @@ TEST_CASE("generalized tx set XDR conversion", "[txset]")
         static_cast<uint32_t>(GENERALIZED_TX_SET_PROTOCOL_VERSION);
     cfg.TESTING_UPGRADE_LEDGER_PROTOCOL_VERSION =
         static_cast<uint32_t>(GENERALIZED_TX_SET_PROTOCOL_VERSION);
+    cfg.TESTING_LEDGER_MAX_SOROBAN_TX_COUNT = 5;
     Application::pointer app = createTestApplication(clock, cfg);
     auto root = TestAccount::createRoot(*app);
     int accountId = 0;
-    auto createTxs = [&](int cnt, int fee) {
+    auto createTxs = [&](int cnt, int fee, bool isSoroban = false) {
         std::vector<TransactionFrameBasePtr> txs;
         for (int i = 0; i < cnt; ++i)
         {
             auto source =
                 root.create("unique " + std::to_string(accountId++),
                             app->getLedgerManager().getLastMinBalance(2));
-            txs.emplace_back(transactionFromOperations(
-                *app, source.getSecretKey(), source.nextSequenceNumber(),
-                {createAccount(getAccount(std::to_string(i)).getPublicKey(),
-                               1)},
-                fee));
+            if (isSoroban)
+            {
+                SorobanResources resources;
+                resources.instructions = 800'000;
+                resources.readBytes = 1000;
+                resources.writeBytes = 1000;
+                resources.extendedMetaDataSizeBytes = 3000;
```

Comment:
NB: when you rebase on top of my most recent changes, this will be changed to `contractEventsSizeBytes` and can be set to 0.